pkg/network/infraconfigurators/bridge_test.go

```diff
@@ -802,6 +803,95 @@ var _ = Describe("Bridge infrastructure configurator", func() {
 			})
 		})
 	})
+
+	Context("DHCP configuration generation", func() {
+		const (
+			ifaceName   = "eth0"
+			launcherPID = 1000
+		)
+
+		var (
+			bridgeConfigurator *BridgePodNetworkConfigurator
+			defaultGwRoute     netlink.Route
+			iface              *v1.Interface
+			podIP              netlink.Addr
+			vmi                *v1.VirtualMachineInstance
+		)
+
+		BeforeEach(func() {
+			iface = v1.DefaultBridgeNetworkInterface()
+			vmi = newVMIWithBridgeInterface("default", "vm1")
+
+			bridgeConfigurator = NewBridgePodNetworkConfigurator(vmi, iface, bridgeIfaceName, launcherPID, handler)
+		})
+
+		When("IPAM is not enabled", func() {
+			createBridgeConfiguratorWithoutIPAM := func() *BridgePodNetworkConfigurator {
+				bc := NewBridgePodNetworkConfigurator(vmi, iface, bridgeIfaceName, launcherPID, handler)
+				bc.podNicLink = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: ifaceName}}
+				return bc
+			}
+
+			BeforeEach(func() {
+				bridgeConfigurator = createBridgeConfiguratorWithoutIPAM()
+			})
+
+			It("should generate a minimal dhcp configuration", func() {
+				expectedDhcpConfig := cache.DHCPConfig{IPAMDisabled: true}
+				Expect(bridgeConfigurator.GenerateNonRecoverableDHCPConfig()).To(Equal(&expectedDhcpConfig))
+			})
+		})
+
+		When("IPAM is enabled", func() {
+			var mac net.HardwareAddr
+
+			createBridgeConfiguratorWithIPAM := func(mac net.HardwareAddr, podIP netlink.Addr) *BridgePodNetworkConfigurator {
+				bc := NewBridgePodNetworkConfigurator(vmi, iface, bridgeIfaceName, launcherPID, handler)
+				bc.podNicLink = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: ifaceName}}
+				bc.vmMac = &mac
+				bc.ipamEnabled = true
+				bc.podIfaceIP = podIP
+				return bc
+			}
+
+			BeforeEach(func() {
+				podIP = netlink.Addr{IPNet: &net.IPNet{IP: net.IPv4(10, 35, 0, 6), Mask: net.CIDRMask(24, 32)}}
+				mac, _ = net.ParseMAC("AF:B3:1F:78:2A:CA")
+				bridgeConfigurator = createBridgeConfiguratorWithIPAM(mac, podIP)
+			})
+
+			When("routes are installed", func() {
+				BeforeEach(func() {
+					defaultGwRoute = netlink.Route{Gw: net.IPv4(10, 35, 0, 1)}
+				})
+
+				BeforeEach(func() {
+					bridgeConfigurator.podIfaceRoutes = []netlink.Route{defaultGwRoute}
+				})
+
+				It("generate a DHCP config also featuring the routes", func() {
+					expectedDhcpConfig := cache.DHCPConfig{
+						IPAMDisabled: false,
+						MAC:          mac,
+						IP:           podIP,
+						Gateway:      defaultGwRoute.Gw,
+					}
+					Expect(bridgeConfigurator.GenerateNonRecoverableDHCPConfig()).To(Equal(&expectedDhcpConfig))
+				})
+			})
+
+			When("the pod does not feature routes", func() {
```

Comment:
This context is optional, it gives no added value. You can just fill the text in the test.